app/src/main/java/com/teamwss/websoso/ui/novelRating/NovelRatingViewModel.kt

```diff
@@ -0,0 +1,287 @@
+                categories = updatedCategories,
+                pastSelectedKeywords = pastSelectedKeywords
+            )
+        )
+    }
+
+    private fun updateCategories(
+        categories: List<KeywordModel.Category>,
+        keyword: KeywordModel.Category.Keyword,
+        isSelected: Boolean
+    ): List<KeywordModel.Category> {
+        return categories.map { category ->
+            val updatedKeywords = category.keywords.map { keywordInCategory ->
+                when (keywordInCategory.keywordId == keyword.keywordId) {
+                    true -> keywordInCategory.copy(isSelected = isSelected)
+                    false -> keywordInCategory
+                }
+            }
+            category.copy(keywords = updatedKeywords)
+        }
+    }
+
+    private fun updateSelectedKeywords(
+        currentSelectedKeywords: List<KeywordModel.Category.Keyword>,
+        keyword: KeywordModel.Category.Keyword,
+        isSelected: Boolean
+    ): List<KeywordModel.Category.Keyword> {
+        return currentSelectedKeywords.toMutableList().apply {
+            when (isSelected) {
+                true -> add(keyword)
+                false -> removeIf { it.keywordId == keyword.keywordId }
+            }
+        }
+    }
+
+    fun initCurrentSelectedKeywords() {
+        val uiState = uiState.value ?: return
+
+        val updatedCategories = uiState.keywordModel.categories.map { category ->
+            val updatedKeywords = category.keywords.map { keyword ->
+                keyword.copy(isSelected = uiState.keywordModel.pastSelectedKeywords.any { it.keywordId == keyword.keywordId })
+            }
+            category.copy(keywords = updatedKeywords)
+        }
+
+        _uiState.value = uiState.copy(
+            keywordModel = uiState.keywordModel.copy(
+                categories = updatedCategories,
+                currentSelectedKeywords = uiState.keywordModel.pastSelectedKeywords
+            )
+        )
+    }
+
+    fun saveSelectedKeywords() {
+        val uiState = uiState.value ?: return
+
+        _uiState.value = uiState.copy(
+            keywordModel = uiState.keywordModel.copy(
+                pastSelectedKeywords = uiState.keywordModel.currentSelectedKeywords
+            )
+        )
+    }
+
+    fun clearKeywordEdit() {
+        val uiState = uiState.value ?: return
+
+        val updatedCategories = uiState.keywordModel.categories.map { category ->
+            val updatedKeywords = category.keywords.map { keyword ->
+                keyword.copy(isSelected = false)
+            }
+            category.copy(keywords = updatedKeywords)
+        }
+
+        _uiState.value = uiState.copy(
+            keywordModel = uiState.keywordModel.copy(
+                categories = updatedCategories,
+                pastSelectedKeywords = emptyList(),
+                currentSelectedKeywords = emptyList()
+            )
+        )
+    }
+
+    fun cancelKeywordEdit() {
```

Comment:
```suggestion
    fun cancelEditingKeyword() {
```
어때여?

다른 곳에서는 `selectedKeyword`처럼 ~한 키워드로 사용한 것 같아서, 이 함수들도 편집 중인 키워드를 취소하다 / 클리어하다 이런 네이밍으로 가져가면 어떨까요?